lib/buildFile.js

```diff
@@ -42,8 +45,49 @@ function buildFile(destination, format, platform, dictionary, filter) {
   if (!fs.existsSync(dirname))
     fs.mkdirsSync(dirname);
 
-  fs.writeFileSync(destination, format(filterProperties(dictionary, filter), platform));
-  console.log(chalk.bold.green('✔︎') + ' ' + destination);
+  let filteredProperties = filterProperties(dictionary, filter);
+
+  // Check for property name Collisions
+  var nameCollisionObj = {};
+  filteredProperties.allProperties && filteredProperties.allProperties.forEach((propertyData) => {
+    let propertyName = propertyData.name;
+    if(!nameCollisionObj[propertyName]) {
+      nameCollisionObj[propertyName] = [];
+    }
+    nameCollisionObj[propertyName].push(propertyData);
+  });
+
+  var PROPERTY_NAME_COLLISION_WARNINGS = GroupMessages.GROUP.PropertyNameCollisionWarnings;
+  var WARNING_COLOR = 'darkorange';
+  var HIGHLIGHT_COLOR = 'orangered';
+  var HELP_COLOR = 'orange';
+  Object.keys(nameCollisionObj).map((propertyName) => {
```

Comment:
Change `.map` here to `.forEach` since we aren't actually mapping the array to some variable.

Reply:
Whoops. Good call.